### PR TITLE
refactor(migrate/eslint): improve naming-convention migration

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This PR improves the migration of the `naming-convention` rule.
It attempts to migrate to the new `conventions` option of `useNamingConvention`.

Unfortunately I encounter stack overflow when I am trying to create regexes.
Thus, I commented all codes that compile a regex.
This makes the migration greatly less useful.

I am open to suggestions to solve this issue.

I also fixed a bug in `RestrictedRegex` that ruled out `([^x])`.

## Test Plan

I extended a test.

### CODE REVIEW

File: crates/biome_js_analyze/tests/specs/style/useNamingConvention/malformedSelector.js.snap [hunk 1]
```diff
@@ -121,7 +121,7 @@ malformedSelector.options:35:21 deserialize ━━━━━━━━━━━━
 ```
 malformedSelector.options:41:21 deserialize ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 
-  × The `global` scope can only be used on type and variable kinds.
+  × The `global ` scope cannot be used with the `class member` kind.
```
Comment (Sec-ant) [inline crates/biome_js_analyze/tests/specs/style/useNamingConvention/malformedSelector.js.snap:124]: `` `global ` `` => `` `global` ``

**Comment by ematipico** [comment]
Personally, I don't think we should provide a migration path to this rule, because regex coming from eslint are JavaScript regex, and we use a crate that doesn't support JavaScript regex. While this is a best effort work, users should be aware of possible discrepancies.

Personally, it's not worth the trouble, but maybe I'm wrong. I hope I am.

I wished you had discussed this before, but the work is done now.

**Comment by Conaclos** [comment]
> I wished you had discussed this before, but the work is done now.

It was a bullet point in my task list in #2770
I thought everyone expected a migration path for the new option.

> Personally, I don't think we should provide a migration path to this rule, because regex coming from eslint are JavaScript regex, and we use a crate that doesn't support JavaScript regex

Yes, however restricted regex should reject regexes that e don't support.

Maybe you are right. This PR already took me too much time. Especially because of the stack overflow.
The resulting migration seems a bit drafty.

**Comment by Conaclos** [comment]
I removed the attempt of translating JavaScript regexes into Rust regexes.

**Pull request merged**